clang/include/clang/AST/TemplateBase.h

```diff
@@ -80,6 +81,13 @@ class TemplateArgument {
     /// that was provided for an integral non-type template parameter.
     Integral,
 
+    /// The template argument is a non-type template argument that can't be
+    /// represented by the special-case Declaration, NullPtr, or Integral
+    /// forms. These values are only ever produced by constant evaluation,
+    /// so cannot be dependent.
+    /// TODO: merge Declaration, NullPtr and Integral into this?
```

Comment:
Do you intend to make a follow up patch?

Reply:
Honestly, no. Not in the near future, at least.

Reply:
That's fine, i wanted to check!